homework_1_decorators/homework_1_decorators.py

```diff
@@ -0,0 +1,52 @@
+import functools
+import requests
+import psutil
+from collections import Counter
+
+
+def memory(f):
+    def internal(*args, **kwargs):
+        process = psutil.Process()
+        memory_usage_before = process.memory_info().rss
+        print(f'Usage memory before {memory_usage_before}')
+        f(*args, **kwargs)
+        memory_usage_after = process.memory_info().rss
+        print(f'Usage memory after {memory_usage_after}')
+    return internal
+        
+
+
+def cache(max_limit):
+    def internal(f):
+        @functools.wraps(f)
+        def deco(*args, **kwargs):
+            cache_key = (args, tuple(kwargs.items()))
+            if cache_key in deco._cache:
+                return deco._cache[cache_key]
+            result = f(*args, **kwargs)
+            if len(deco._cache) >= max_limit:
+                least_used_key = min(deco._cache, key=deco._cache.get)
+                del deco._cache[least_used_key]
+            deco._cache[cache_key] = result
+            return result
+        deco._cache = Counter()
```

Comment:
Here wrong logic, Counter is bad here:
Check pls this screen 

![image](https://github.com/AlinaKovban/hillel_python_pro/assets/17034253/5fa0ebf3-9adb-435f-9726-7baca54d0a84)